projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts

```diff
@@ -0,0 +1,411 @@
+  @Input() get breadcrumbs(): LfTreeNode[] {
+    return this._breadcrumbs;
+  }
+
+  /**
+   * function to initialize the lf-file-explorer component
+   * @param provider LfRepositoryService service
+   * @param selectedNode the id of the node to select, or a Entry starting from the selected entry
+  */
+  @Input() initAsync = async (treeNodeService: LfTreeNodeService, selectedNode?: string | LfTreeNode): Promise<void> => {
+    await this.zone.run(async () => {
+      try {
+        this.treeNodeService = treeNodeService;
+      } catch (error) {
+        console.error(error);
+        this.hasError = true;
+        return;
+      }
+      await this.initializeAsync(selectedNode);
+    });
+  };
+  @Input() isSelectable?: (treeNode: LfTreeNode) => Promise<boolean>;
+  @Input() multiple: boolean = false;
+  @Input()
+  setSelectedValuesAsync: (valuesToSelect: LfTreeNode[]) => Promise<void> = async (valuesToSelect: LfTreeNode[]) => {
+    const selectableValues = await this.mapTreeNodesToLfSelectableAsync(valuesToSelect);
+    if (this.entryList == null) {
+      setTimeout(() => {
+        this.setSelectedValuesAsync(valuesToSelect);
+      });
+      return;
+    }
+    this.entryList.setSelectedValuesAsync(selectableValues, this.checkForMoreDataCallback.bind(this)).then((selected: ILfSelectable[]) => {
+      this.entrySelected.emit(this.convertSelectedItemsToTreeNode(selected));
+    });
+  };
+ 
+  @Output() entrySelected = new EventEmitter<LfTreeNode[] | undefined>();
+
+  /** @internal */
+  currentFolderChildren: ILfSelectable[] = [];
+  /** @internal */
+  hasError: boolean = false;
+  /** @internal */
+  isLoading: boolean = false;
+  /** @internal */
+  nextPage: string | undefined;
+  /** @internal */
+  treeNodeService!: LfTreeNodeService;
+
+  /** @internal */
+  readonly OPEN = this.localizationService.getStringObservable('OPEN');
+  /** @internal */
+  readonly AN_ERROR_OCCURED = this.localizationService.getStringObservable('AN_ERROR_OCCURED');
+  /** @internal */
+  get shouldShowErrorMessage(): boolean {
+    return this.hasError;
+  }
+  /** @internal */
+  readonly THIS_FOLDER_IS_EMPTY = this.localizationService.getStringObservable('THIS_FOLDER_IS_EMPTY');
+  /** @internal */
+  get shouldShowEmptyMessage(): boolean {
+    return this.currentFolderChildren.length === 0 && !this.hasError;
+  }
+
+  /** @internal */
+  protected _currentEntry?: LfTreeNode;
+  /** @internal */
+  // protected selectable: Selectable = new Selectable();
+
+  /** @internal */
+  private _breadcrumbs: LfTreeNode[] = [];
+  /** @internal */
+  private scrolledIndexChanged = new Subject();
+  
+  /** @internal */
+  constructor(
+    /** @internal */
+    public ref: ChangeDetectorRef,
+    /** @internal */
+    public popupDialog: MatDialog,
+    /** @internal */
+    public zone: NgZone,
+    /** @internal */
+    public localizationService: AppLocalizationService
+  ) {
+    this.scrolledIndexChanged.pipe(debounceTime(200)).subscribe(async () => {
```

Comment:
Sometimes this get called twice. Is the debounceTime(200) an attempt to fix that behavior? Why is it 200?